libqtile/widget/textbox.py

```diff
@@ -28,14 +28,22 @@
 
 
 class TextBox(base._TextBox):
-    """A flexible textbox that can be updated from bound keys, scripts, and qshell"""
+    """A flexible textbox that can be updated from bound keys, scripts, and qshell
+
+    Callback functions can be assigned to button presses by passing a dict to the
+    'callbacks' kwarg. For example: {1: func} will execute func when the widget receives
+    a button 1 press. The Qtile instance of passed as the only argument to the callback
+    functions.
+
+    """
     orientations = base.ORIENTATION_HORIZONTAL
     defaults = [
         ("font", "sans", "Text font"),
         ("fontsize", None, "Font pixel size. Calculated if None."),
         ("fontshadow", None, "font shadow color, default is None(no shadow)"),
         ("padding", None, "Padding left and right. Calculated if None."),
         ("foreground", "#ffffff", "Foreground colour."),
+        ("callbacks", {}, "Dict of button press callback functions."),
```

Comment:
Can we use the same language as for Mouse() configuration, i.e. "Button1" instead of 1, "Button2" instead of 2, etc.?

And maybe name this mouse_callbacks? Other than that lgtm.

Reply:
Done!